GitUpKit/Components/GIDiffContentsViewController.m

```diff
@@ -332,7 +340,13 @@ - (void)_reloadDeltas {
           GCDiffPatch* patch = [self.repository makePatchForDiffDelta:delta isBinary:&isBinary error:&error];
           if (patch) {
             XLOG_DEBUG_CHECK(!isBinary || patch.empty);
-            if (patch.empty) {
+
+            BOOL isImage = [[NSImage alloc] initWithContentsOfFile:[self.repository absolutePathForFile:delta.canonicalPath]] != nil;
```

Comment:
I think we need a lighter weight approach for determining if a path may be an image. If we have a lot of images and large images then we're going to load them all into memory up front and that could be very slow.

Reply:
Great point!
I switched it over to using [UTIs](https://developer.apple.com/documentation/uniformtypeidentifiers?language=objc), which should be very fast and still plenty reliable 🙂 (c4f52a4fbc19dd030341c6b220c7bf75d31b5216)

I'm not super familiar with memory management bridging for core foundation entities, though, so that part could definitely use a thorough look 😅 (the same goes for my other fix).

Reply:
I think that should mostly be right, but to verify NSImageView can load it we probably should use `NSImage.imageTypes` to get the array of valid UTIs and then check conformance against those to see if we match any. I think `kUTTypeImage` type might be too generic.

Reply:
Sorted! bae1fc2d943165e92308ef3d57e030c3abd54e96